engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -101,13 +104,13 @@ public T getValue() {
     }
 
     public boolean setValue(T newValue) {
-        if (!validator.validate(newValue))
+        if (!validator.validate(newValue)) {
+            LOGGER.warn("The passed value {} is invalid.", newValue);
```

Comment:
That's what I prefer too. But from your examples I'm just going to rename `validate` to `validateWithWarnings` and `fastValidate` to `validate`.